app/uk/gov/hmrc/vatsubscription/services/UpdateReturnPeriodService.scala

```diff
@@ -41,7 +41,12 @@ class UpdateReturnPeriodService @Inject()(updateVatSubscriptionConnector: Update
   def constructReturnPeriodUpdateModel(updatedReturnPeriod: ReturnPeriod, welshIndicator: Boolean)
                                       (implicit user: User[_]): UpdateVatSubscription = {
 
-    val agentOrCapacitor: Option[AgentOrCapacitor] = user.arn.map(AgentOrCapacitor(_, None))
+    val agentContactDetails: Option[ContactDetails] = Some(ContactDetails(None, None, None,
+      updatedReturnPeriod.transactorOrCapacitorEmail, None))
+
+    val agentOrCapacitor: Option[AgentOrCapacitor] = user.arn.map(AgentOrCapacitor(_, agentContactDetails))
```

Comment:
If we have no agent email this will write an empty comms obkect thus:

"commDetails": {
				
			},

This is strictly OK int the schema but I wouldn't be surprised if ETMP fail it.

In any case we are writing an incomplete and unecessary section

The code would work ok here for instance if the val was changed soemthing like this:

_val agentContactDetails: Option[ContactDetails] = if(updatedReturnPeriod.agentOrCapcitorEmail.isDefined) Some(ContactDetails(None, None, None,
updatedReturnPeriod.transactorOrCapacitorEmail, None) else None_